components/constellation/constellation.rs

```diff
@@ -1327,30 +1327,27 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
     }
 
     fn handle_frame_size_msg(&mut self,
-                             pipeline_id: PipelineId,
-                             size: &TypedSize2D<f32, PagePx>) {
-        let msg = ConstellationControlMsg::Resize(pipeline_id, WindowSizeData {
-            visible_viewport: *size,
-            initial_viewport: *size * ScaleFactor::new(1.0),
-            device_pixel_ratio: self.window_size.device_pixel_ratio,
-        }, WindowSizeType::Initial);
-
-        // Store the new rect inside the pipeline
-        let result = {
-            // Find the pipeline that corresponds to this rectangle. It's possible that this
-            // pipeline may have already exited before we process this message, so just
-            // early exit if that occurs.
-            match self.pipelines.get_mut(&pipeline_id) {
+                             iframe_sizes: Vec<(PipelineId, TypedSize2D<f32, PagePx>)>) {
+        for (pipeline_id, size) in iframe_sizes {
+            let result = match self.pipelines.get_mut(&pipeline_id) {
                 Some(pipeline) => {
-                    pipeline.size = Some(*size);
-                    pipeline.event_loop.send(msg)
+                    if pipeline.size != Some(size) {
+                        pipeline.size = Some(size);
+                        let msg = ConstellationControlMsg::Resize(pipeline_id, WindowSizeData {
+                            visible_viewport: size,
+                            initial_viewport: size * ScaleFactor::new(1.0),
+                            device_pixel_ratio: self.window_size.device_pixel_ratio,
+                        }, WindowSizeType::Initial);
+                        Some(pipeline.event_loop.send(msg))
+                    } else {
+                        None
+                    }
                 }
-                None => return,
+                None => None
+            };
+            if let Some(Err(e)) = result {
```

Comment:
That's actually not possible, the compiler complains about `self` being mutably borrowed in the match line; because `self` needs to be mutably borrowed again for `handle_send_error`. I had to resort to using this method then :)